File: frontend/src/types/post.ts
export interface Post{
    id: string;
    content: string;

    like_count: number;
    comment_count: number;
    share_count: number;

    created_by: string;
    created_at: string;

    images: PostImage[];
}

export interface PostImage{
    id: string;
    post_id: string;
    image_url: string;
}

File: app/schemas/post.py
from pydantic import BaseModel
from uuid import UUID
from datetime import datetime
from .user import UserProfile

class PostImageSchema(BaseModel):
    id: UUID
    image_url: str
    post_id: UUID
    
class PostSchema(BaseModel):
    id: UUID
    content: str | None
    
    like_count: int
    comment_count: int
    share_count: int
    
    created_at: datetime
    
    creator: UserProfile
    
    images: list[PostImageSchema] = []
    